dio/lib/src/dio_error.dart

```diff
@@ -2,35 +2,109 @@ import 'options.dart';
 import 'response.dart';
 
 enum DioErrorType {
-  /// It occurs when url is opened timeout.
-  connectTimeout,
+  /// Caused by a connection timeout.
+  connectionTimeout,
```

Comment:
The changes with `DioErrorType` should be considered as a breaking change, which means you'll need to update the migration guide?

Reply:
Yes, this is a breaking change

Reply:
I don't think I need to add anything to the migration guide. From a user's point of view, nothing changes. The user can still catch exceptions the same way as before. All breaking changes are done from within the package in this PR.